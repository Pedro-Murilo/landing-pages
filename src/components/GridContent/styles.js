import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacing.sizes.huge} 0;
  `}
`;
