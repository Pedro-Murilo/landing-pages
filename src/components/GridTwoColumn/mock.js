export default {
  title: 'Grid Two Column',
  text: 'Lorem',
  srcImg: 'assets/images/logo.svg',
};
