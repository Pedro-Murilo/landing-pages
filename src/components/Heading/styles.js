import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.xmedium};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.big};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.big};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, darkTheme, size, uppercase }) => css`
    color: ${darkTheme ? theme.colors.red : theme.colors.green};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
