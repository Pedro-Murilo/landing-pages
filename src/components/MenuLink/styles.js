import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacing.sizes.small};
    color: ${theme.colors.green};
    position: relative;

    &::after {
      content: '';
      position: absolute;
      bottom: 0.75rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.red};
      transition: all 0.3s ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
