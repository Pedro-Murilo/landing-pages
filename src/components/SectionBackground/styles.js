import styled, { css } from 'styled-components';

const isBackgroundActive = (theme) => css`
  background: ${theme.colors.black};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.black};
    ${background && isBackgroundActive(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
