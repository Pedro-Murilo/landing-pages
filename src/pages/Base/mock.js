import linksMock from '../../components/NavLink/mock';

export default {
  links: linksMock,
  logoData: {
    text: 'Logo',
    link: '#',
  },
  footerHtml: '<p>Test footer</p>',
};
