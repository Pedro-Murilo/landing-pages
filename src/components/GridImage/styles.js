import styled, { css } from 'styled-components';

import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacing.sizes.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.sizes.xmedium};
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${() => css`
    width: 100%;
    transition: all 200ms ease-in;

    &:hover {
      transform: scale(1.1) rotate(5deg);
    }
  `}
`;
