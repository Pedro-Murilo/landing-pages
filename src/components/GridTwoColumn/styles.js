import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacing.sizes.medium};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: ${theme.spacing.sizes.xmedium};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacing.sizes.medium};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${() => css``}
`;

export const Image = styled.img`
  ${() => css``}
`;
