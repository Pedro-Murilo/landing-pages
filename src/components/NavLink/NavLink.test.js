import { renderTheme } from '../../styles/RenderTheme';
import { screen } from '@testing-library/react';

import { NavLink } from '.';
import mock from './mock';
import { theme } from '../../styles/theme';

describe('<NavLink />', () => {
  it('should render links', () => {
    renderTheme(<NavLink links={mock} />);
    expect(screen.getAllByRole('link')).toHaveLength(mock.length);
  });

  it('should not render links', () => {
    renderTheme(<NavLink />);
    expect(screen.queryAllByText(/links/i)).toHaveLength(0);
  });

  it('should render links', () => {
    renderTheme(<NavLink links={mock} />);
    expect(screen.getByText(/link 5/i).parentElement).toHaveStyleRule(
      'flex-flow',
      'column wrap',
      {
        media: theme.media.lteMedium,
      },
    );
  });

  it('should match snapshot', () => {
    renderTheme(<NavLink links={mock} />);
    expect(screen.getAllByRole('link')).toMatchSnapshot();
  });
});
