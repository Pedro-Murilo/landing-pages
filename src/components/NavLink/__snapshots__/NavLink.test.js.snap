// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> should match snapshot 1`] = `
Array [
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  color: #06cf32;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #e34040;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target1"
    target="_self"
  >
    Link 1
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  color: #06cf32;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #e34040;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target2"
    target="_self"
  >
    Link 2
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  color: #06cf32;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #e34040;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target3"
    target="_self"
  >
    Link 3
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  color: #06cf32;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #e34040;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target4"
    target="_self"
  >
    Link 4
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  color: #06cf32;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #e34040;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target5"
    target="_self"
  >
    Link 5
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  color: #06cf32;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #e34040;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target6"
    target="_self"
  >
    Link 6
  </a>,
]
`;
